internal/resource_identity_provider.go

```diff
@@ -0,0 +1,250 @@
+				Description: "The OIDC discovery URL of the identity provider.",
+				Required:    true,
+			},
+			"user_id_claim": schema.StringAttribute{
+				Description: "The claim used to identify the user in the identity provider's token.",
+				Required:    true,
+			},
+			"audiences": schema.ListAttribute{
+				Required:    true,
+				ElementType: types.StringType,
+			},
+			"issuers": schema.ListAttribute{
+				Description: "List of issuers for the identity provider.",
+				Required:    true,
+				ElementType: types.StringType,
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (resource *identityProviderResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var plan identityProviderResourceModel
+	diags := request.Plan.Get(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	registration, diags := toBoxerIssuerModel(ctx, plan)
+	if diags.HasError() {
+		response.Diagnostics.Append(diags...)
+		return
+	}
+	err := resource.issuerClient.PostProvider(ctx, registration, issuer.PostProviderParams{ID: plan.Name.ValueString()})
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error Creating Identity Provider",
+			"An error occurred while creating the identity provider: "+err.Error(),
+		)
+		return
+	}
+	plan.ID = types.StringValue(plan.Name.ValueString())
+	response.State.Set(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (resource *identityProviderResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state identityProviderResourceModel
+	diags := request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	apiData, err := resource.issuerClient.GetProvider(ctx, issuer.GetProviderParams{ID: state.Name.ValueString()})
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error reading identity provider",
+			"An error occurred while reading the identity provider: "+err.Error(),
+		)
+		return
+	}
+	state = fromBoxerIssuerModel(state.ID.ValueString(), apiData)
+	diags = response.State.Set(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (resource *identityProviderResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var plan identityProviderResourceModel
+	diags := request.Plan.Get(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	var state identityProviderResourceModel
+	diags = request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
```

Comment:
```suggestion
	response.Diagnostics.Append(diags...)
```
you can avoid duplicating this code by using `defer` at the start of the method:

```
defer func() { 
        response.Diagnostics.Append(diags...)
	if response.Diagnostics.HasError() {
		return
	}
}
```